amespahdbpythonsuite/parsers.py

```diff
@@ -0,0 +1,543 @@
+            `scale`, `intensity`, `symmetry`.
+        """
+
+        def geometry_keys():
+            """Returns the XML tags for the children of <geometry> as a
+            dictionary."""
+            geo_keys = {
+                'position': int,
+                # 'uid': int,
+                # 'type': int,
+                'x': float,
+                'y': float,
+                'z': float,
+            }
+            return geo_keys
+
+        def transition_keys():
+            """Returns the XML tags for the children of <transitions> as a
+            dictionary."""
+            tran_keys = {
+                'frequency': float,
+                # 'uid': int,
+                'scale': float,
+                'intensity': float,
+                'symmetry': str,
+            }
+            return tran_keys
+
+        def laboratory_keys():
+            """Returns the XML tags for the children of <laboratory> as a
+            dictionary."""
+            lab_keys = {
+                'frequency': float,
+                'intensity': float,
+            }
+            return lab_keys
+
+        def parse_atom_to_dict(atom, namespace, geometry_keys):
+            """Parse a single <atom> XML element to dictionary.
+
+            Args:
+                atom: XML element of type <atom> to be parsed.
+                namespace (str): XML namespace.
+                geometry_keys (dict): Dict of expected keys/dtypes.
+
+            Returns:
+                atom_dict (dict): Dict of atom properties/quantities.
+            """
+            atom_properties = atom.getchildren()
+            atom_dict = {}
+
+            for atom_property in atom_properties:
+                # Parse children.
+                key = atom_property.tag.split(namespace)[-1]
+                value = atom_property.text
+
+                if key == 'type':
+                    value = int(value)
+                else:
+                    # Cast to desired type (key dependent).
+                    desired_type = geometry_keys[key]
+                    if not isinstance(value, desired_type):
+                        value = desired_type(value)
+
+                # Store in new dictionary.
+                atom_dict[key] = value
+
+            return atom_dict
+
+        def parse_mode_to_dict(mode, namespace, transition_keys):
+            """Parse a single <mode> XML element to dictionary.
+
+            Args:
+                mode: XML element of type <mode> to be parsed.
+                namespace (str): XML namespace.
+                transition_keys (dict): Dict of expected keys/dtypes.
+
+            Returns:
+                mode_dict (dict): Dict of mode properties/quantities.
+            """
+            mode_properties = mode.getchildren()
+            mode_dict = {}
+
+            for mode_property in mode_properties:
+                key = mode_property.tag.split(namespace)[-1]
+                value = mode_property.text  # of type str.
```

Comment:
Type 'str' is not numeric; the method name is therefore misleading ...

Reply:
The data is numeric but LXML returns the 'text' field (containing the data) as a string; shortly afterward it is cast to its appropriate numerical data type (3 lines below).